src/pretix/presale/views/__init__.py

```diff
@@ -112,16 +112,28 @@ def keyfunc(pos):
                 or self.request.event.settings.attendee_emails_asked
                 or pos_additional_fields.get(pos.pk)
             )
+
             addon_penalty = 1 if pos.addon_to_id else 0
-            if downloads or pos.pk in has_addons or pos.addon_to_id:
-                return i, addon_penalty, pos.pk, 0, 0, 0, 0, (pos.subevent_id or 0), (pos.seat_id or 0)
-            if answers and (has_attendee_data or pos.item.questions.all()):
-                return i, addon_penalty, pos.pk, 0, 0, 0, 0, (pos.subevent_id or 0), (pos.seat_id or 0)
 
+            if downloads \
+                    or pos.pk in has_addons \
+                    or pos.addon_to_id \
+                    or pos.item.issue_giftcard \
+                    or answers and (has_attendee_data or pos.item.questions.exists()):
```

Comment:
```suggestion
                    or (answers and (has_attendee_data or pos.item.questions.exists())):
```
Nobody can remember the precedence of or and and :see_no_evil: